### DISCUSSION TITLE
【前端路由】

### DISCUSSION THREAD

**Discussion by swustzzh** [swustzzh/coder_books#28] [General]
## 单页面|多页面
页面的显示是靠多个HTML还是靠JS来切换内容

- 多页面
  - 向服务器请求新的HTML文件来切换显示新的内容
- 单页面 
  - 靠一个HTML，通过JS来控制这个HTML中应该显示什么内容
  - 单页面应用是不需要靠请求服务器来返回新的HTML内容的
  - 路由是什么？
    1. URL路由标识
      - hash模式：
        - \#是一种URL的截断
        - \#及后面的内容是浏览器端识别的内容
        - \#前面的内容才是真正向服务器请求的URL
      - history模式 : RestFul的路由模式
    2. 控制单页面显示什么内容

**Comment by swlws** [comment]
多页面有点狭隘了，分几种场景：

1. 全部由 html 文件文件组件
   1.1 前后端分离的项目，通常由前段自己控制。
   1.2 JSP、PHP等项目，一般由后台返回编译后的文件，或者返回路由，由前段控制跳转
2. 每一个 html，都是单文件项目
   1.1 工程路由通常为固定路由，前后端都可以控制
   1.1 工程内的路由，通常由前段自己控制，比如 vue-router、react-router

**Reply by swustzzh** [reply]
vue-router,react-router不是为了处理单页面应用嘛？

**Reply by swlws** [reply]
多页应用，每个页面也可以是单文件工程哇

**Reply by swustzzh** [reply]
多页面本质上还是多个html，具体每个页面怎么处理，可能就看不同页面的设计了
